src/ess/powder/calibration.py

```diff
@@ -94,18 +94,19 @@ def to_cif_format(self) -> sc.DataArray:
 
 
 def assemble_output_calibration(
-    beamline: CalibratedBeamline[SampleRun],
+    data: DspacingData[SampleRun],
```

Comment:
This doesn't seem quite right.

Inside this function, you are only using static coordinates `Ltotal` and `two_theta`.
If you use `DspacingData` instead of `CalibratedBeamline`, it means that when you are streaming live data, you would recompute these positions every time some new events come in,

We want to avoid this by using `CalibratedBeamline` instead.

Reply:
I did this to support processing multiple banks. I wanted to have a single calibration table which requires access to all banks at the same time. But `CalibratedBeamline` is mapped over banks. The earliest value that is merged is `DspacingData`.

However, I am going to change it back and not merge the banks and instead process them separately. Then `assemble_output_calibration` can be reverted.

Reply:
Reverted